src/hssm/rl/likelihoods/builder.py

```diff
@@ -0,0 +1,491 @@
+"""The log-likelihood function for the RLDM model."""
+
+import functools
+from dataclasses import dataclass
+from typing import Any, Callable, Protocol
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+from jax.lax import scan
+from pytensor.graph import Op
+
+from hssm.distribution_utils.func_utils import make_vjp_func
+from hssm.distribution_utils.jax import make_jax_logp_ops
+from hssm.distribution_utils.onnx import make_jax_matrix_logp_funcs_from_onnx
+
+
+class AnnotatedFunction(Protocol):
```

Comment:
Now that we want to be type-safe, we should make protocols for learning process and decision process functions with different metatdata requirements